torch/backends/cudnn/rnn.py

```diff
@@ -190,7 +196,11 @@ def forward(fn, input, hx, weight, output, hy):
         handle = cudnn.get_handle()
         fn.datatype = cudnn._typemap[input.type()]
         is_input_packed = fn.batch_sizes is not None
-
+        if is_input_packed:
+            warnings.warn(
+                "Warning: persistent algorithm not supported for variable length input."
+                "Switching to standard")
+            fn.persistent = False  # persistent algo is not supported for variable length input
```

Comment:
Ok, and what do you suggest if benchmark mode is not enabled? Have a crude heuristic, or always default to standard? Another question - is there a way to query compute capability from python? Persistent is available on Pascal only. 
I'm somewhat afraid of benchmarking overhead, esp if sequences with very different lengths or batchsizes are passed through. And binning with size 50 does seem arbitrary, but we can start with that, I guess.